src/StackAdmin/Azs.Compute.Admin/Module/Azs.Compute.Admin/Generated.PowerShell.Commands/SwaggerPathCommands/Get-AzsDisk.ps1

```diff
@@ -0,0 +1,201 @@
+<#
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License. See License.txt in the project root for license information.
+
+Code generated by Microsoft (R) PSSwagger
+Changes may cause incorrect behavior and will be lost if the code is regenerated.
+#>
+
+<#
+.SYNOPSIS
+    Get or list managed disks properties.
+
+.DESCRIPTION
+    Returns a list of disks.
+
+.PARAMETER ResourceId
+    The resource id.
+
+.PARAMETER InputObject
+    The input object of type Microsoft.AzureStack.Management.Compute.Admin.Models.Disk.
+
+.PARAMETER Location
+    Location of the resource.
+
+.PARAMETER Start
+    The start index of disks in query.
+
+.PARAMETER SharePath
+    The source share which the resource belongs to.
+
+.PARAMETER Count
+    The maximum number of disks to return.
+
+.PARAMETER UserSubscriptionId
+    Tenant Subscription Id which the resource belongs to.
+
+.PARAMETER Status
+    The parameters of disk state.
+
+.PARAMETER Name
+    The disk guid as identity.
+
+.EXAMPLE
+
+	PS C:\> $disks = Get-AzsDisk -location local
+    PS C:\> $disk = Get-AzsDisk -location local -name $disks[0].DiskId
+#>
+function Get-AzsDisk
+{
+    [OutputType([Microsoft.AzureStack.Management.Compute.Admin.Models.Disk])]
+    [CmdletBinding(DefaultParameterSetName='Disks_List')]
+    param(    
+        [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true, ParameterSetName = 'ResourceId_Disks_Get')]
+        [System.String]
+        $ResourceId,
+    
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true, ParameterSetName = 'InputObject_Disks_Get')]
+        [Microsoft.AzureStack.Management.Compute.Admin.Models.Disk]
+        $InputObject,
+    
+        [Parameter(Mandatory = $false, ParameterSetName = 'Disks_List')]
+        [Parameter(Mandatory = $false, ParameterSetName = 'Disks_Get')]
+        [System.String]
+        $Location,
+    
+        [Parameter(Mandatory = $false, ParameterSetName = 'Disks_List')]
+        [System.Nullable`1[System.Int32]]
+        $Start = $null,
+    
+        [Parameter(Mandatory = $false, ParameterSetName = 'Disks_List')]
+        [System.String]
+        $SharePath,
+    
+        [Parameter(Mandatory = $false, ParameterSetName = 'Disks_List')]
+        [System.Nullable`1[System.Int32]]
+        $Count = $null,
+    
+        [Parameter(Mandatory = $false, ParameterSetName = 'Disks_List')]
+        [System.String]
+        $UserSubscriptionId,
+    
+        [Parameter(Mandatory = $false, ParameterSetName = 'Disks_List')]
```

Comment:
Can users actually set this on the client side?  I remember a conversation about this but cannot remember the answer.

Reply:
Yes, they could. We will return the disk which meets the status.